File: vue-typescript/src/CustomComponent.ts
import Vue from 'vue';

export default Vue.extend({
  template: '<button v-on:click="handleClick">CLICK ME ({{count}})</button>',
  data() {
    return {
      count: 0
    }
  },
  methods: {
    handleClick() {
      this.count += 1;
    }
  }
});

File: react-kotlin/src/main/kotlin/CustomComponent.kt
import React.createElement

class CustomComponentState(val count: Int) {}

class CustomComponent(props: Any): React.Component<Any, CustomComponentState>(props) {
  init {
    state = CustomComponentState(0)
  }

  fun handleClick() {
    setState { state, _ -> CustomComponentState(state.count + 1) }
  }

  override fun render(): dynamic {
    return createElement(
      "button",
      object {
        @JsName("onClick")
        val onClick = this@CustomComponent::handleClick
      },
      "CLICK ME (", state.count, ")")
  }
}
